common/models/acl.js

```diff
@@ -506,12 +506,19 @@ module.exports = function(ACL) {
   };
 
   ACL.resolveRelatedModels = function() {
-    if (!this.roleModel) {
-      var reg = this.registry;
-      this.roleModel = reg.getModelByType(loopback.Role);
-      this.roleMappingModel = reg.getModelByType(loopback.RoleMapping);
-      this.userModel = reg.getModelByType(loopback.User);
-      this.applicationModel = reg.getModelByType(loopback.Application);
+    var self = this;
+    if (!self.roleModel) {
+      var reg = self.registry;
+      var builtInModels = {
+        roleModel: reg.getModel('Role'),
+        roleMappingModel: reg.getModel('RoleMapping'),
+        userModel: reg.getModel('User'),
+        applicationModel: reg.getModel('Application'),
+      };
+
+      for (var m in builtInModels) {
+        self[m] = reg.getModelByType(builtInModels[m]);
```

Comment:
[the first call](https://github.com/strongloop/loopback/pull/2628#discussion-diff-74943161R513) is to resolve the builtin models (e.g. like the old code did by `loopback.Role`) but from the registry that `ACL` is registered to.
[the second call](https://github.com/strongloop/loopback/pull/2628#discussion-diff-74943161R520) is to resolve any custom model that might extend the builtins.
`getModelByType()` calls `getModel()` internally only when called with string name argument. Otherwise it compares registered model's prototypes with the model provided as argument